### ISSUE TITLE
Test failed: External ArrayBuffer with finalizer

### ISSUE THREAD

**Issue by mhdawson** [nodejs/node-addon-api#735]
Test failed: External ArrayBuffer with finalizer seems to have been failing since May 16, on Windows for master and 14.x
```text
Running test 'objectreference'
Running test 'version_management'

All tests passed!
Test failed: External ArrayBuffer with finalizer
C:\workspace\node-test-node-addon-api-new\nodes\win-vs2019\node-addon-api\test\testUtil.js:17
        throw e;
        ^

AssertionError [ERR_ASSERTION]: Expected values to be strictly equal:

1 !== 0

    at Array.<anonymous> (C:\workspace\node-test-node-addon-api-new\nodes\win-vs2019\node-addon-api\test\arraybuffer.js:42:14)
    at runGCTests (C:\workspace\node-test-node-addon-api-new\nodes\win-vs2019\node-addon-api\test\testUtil.js:14:17)
    at Immediate._onImmediate (C:\workspace\node-test-node-addon-api-new\nodes\win-vs2019\node-addon-api\test\testUtil.js:21:9)
    at processImmediate (internal/timers.js:456:21) {
  generatedMessage: true,
  code: 'ERR_ASSERTION',
  actual: 1,
  expected: 0,
  operator: 'strictEqual'
}
npm ERR! Test failed.  See above for more details.
Build step 'Conditional steps (multiple)' marked build as failure
```

**Comment by mhdawson** [comment]
Don't see any changes to the test files for arraybuffer.js or arraybuffer.cc recently.

@gabrielschulhof are you aware of any finalizer related changes on the node core side recently?

**Comment by mhdawson** [comment]
This looks like it could be related: https://github.com/nodejs/node/pull/33321.  It also went in to 14.x as well so that would explain it failing on both master and 14 but not earlier releases.

**Comment by mhdawson** [comment]
From this comment likely an issue with timing/windows https://github.com/nodejs/node/pull/33321#issuecomment-626253489

as it sounds like the test needed to be updated for core for Windows.

**Comment by gabrielschulhof** [comment]
A fix for this has landed in #737.
